tests/phpunit/includes/polylang/settings-context.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+class PLL_Settings_Context extends PLL_Admin_Context {
+
+	/**
+	 * Gets the context class name.
+	 *
+	 * @since 3.6
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return PLL_Settings::class;
+	}
+
+	/**
+	 * Returns the model according to the context.
+	 * PLL_Admin_Model for Settings.
+	 *
+	 * @since 3.6
+	 *
+	 * @param array $options Polylang options.
+	 * @return PLL_Model
+	 */
+	protected function get_model( array $options ): PLL_Model {
```

Comment:
Shouldn't the type be the same as `PLL_Context->get_model()` ?

Reply:
Since `PLL_Admin_Model` extends `PLL_Model`, we can do this :)

Reply:
Yes, I did that at first because of that, but my IDE didn't like it: 

**_Return type declaration must be compatible with PLL_Context->get_model(options: array) : \PLL_Model. This is only allowed starting with PHP 7.4._** 

But maybe it doesn't notice the inheritance between these two classes.

Reply:
Oh, I didn't know that, it's a php 7.4 feature!
Forget my suggestion then.